cumulusci/robotframework/Salesforce.py

```diff
@@ -84,98 +85,38 @@ def execute(driver_command, params=None):
-                exception = e
-        except WebDriverException as e:
-            if "Other element would receive the click" in str(e):
-                if retry is True:
-                    retry_call = True
-                else:
-                    exception = e
-            else:
-                exception = e
-        except ElementNotInteractableException as e:
-            if retry is True:
-                retry_call = True
-            else:
-                exception = e
-        except Exception as e:
-            exception = e
-
-        if exception:
-            self.selenium.capture_page_screenshot()
-            if self.debug:
-                self.selenium.log_source()
-                from cumulusci.robotframework.utils import set_pdb_trace
-
-                set_pdb_trace()
-                raise e
-            else:
-                raise e
-
-        if retry_call:
-            self.builtin.log(
-                "Retrying call to method {}".format(method_name), level="WARN"
-            )
-            return self._call_selenium(method_name, False, *args, **kwargs)
-
-        return result
-
+    @selenium_retry
     def click_modal_button(self, title):
         locator = lex_locators["modal"]["button"].format(title)
-        self._call_selenium("_click_modal_button", True, locator)
-
-    def _click_modal_button(self, locator):
-        self.selenium.set_focus_to_element(locator)
-        button = self.selenium.get_webelement(locator)
-        button.click()
+        self.selenium.click_button(locator)
 
+    @selenium_retry
     def click_object_button(self, title):
         locator = lex_locators["object"]["button"].format(title)
-        self._call_selenium("_click_object_button", True, locator)
-
-    def _click_object_button(self, locator):
-        button = self.selenium.get_webelement(locator)
-        button.click()
+        self.selenium.click_link(locator)
         self.wait_until_modal_is_open()
 
+    @selenium_retry
     def click_related_list_button(self, heading, button_title):
         locator = lex_locators["record"]["related"]["button"].format(
             heading, button_title
         )
-        self._call_selenium("_click_related_list_button", True, locator)
-
-    def _click_related_list_button(self, locator):
-        self.selenium.set_focus_to_element(locator)
-        self.selenium.get_webelement(locator).click()
+        self.selenium.click_link(locator)
         self.wait_until_modal_is_open()
 
+    @selenium_retry
+    def close_modal(self):
+        """ Closes the open modal """
+        locator = lex_locators["modal"]["close"]
+        self.selenium.click_button(locator)
+
+    @selenium_retry
+    def current_app_should_be(self, app_name):
+        """ Validate the currently selected Salesforce App """
+        locator = lex_locators["app_launcher"]["current_app"].format(app_name)
+        elem = self.selenium.get_webelement(locator)
+        assert app_name == elem.text
```

Comment:
This raises a bare AssertionError which obscures the reason for the error. Add a message?

    assert app_name == elem.text, "Expected app to be {} but found {}".format(app_name, elem.text)